pkg/webhook/managedresource/validatingadmissionpolicy.go

```diff
@@ -42,58 +42,30 @@ func mutateValidatingAdmissionPolicy(vap *admv1.ValidatingAdmissionPolicy, isHub
 				{
 					RuleWithOperations: admv1.RuleWithOperations{
 						Rule: admv1.Rule{
-							APIGroups:   []string{""},
-							Resources:   []string{"namespaces"},
-							APIVersions: []string{"v1"},
-						},
-						Operations: []admv1.OperationType{admv1.Create, admv1.Update, admv1.Delete},
-					},
-				},
-				{
-					RuleWithOperations: admv1.RuleWithOperations{
-						Rule: admv1.Rule{
-							APIGroups:   []string{""},
-							Resources:   []string{"resourcequotas"},
-							APIVersions: []string{"v1"},
-						},
-						Operations: []admv1.OperationType{admv1.Create, admv1.Update, admv1.Delete},
-					},
-					ResourceNames: []string{"default"},
-				},
-				{
-					RuleWithOperations: admv1.RuleWithOperations{
-						Rule: admv1.Rule{
-							APIGroups:   []string{"networking.k8s.io"},
-							Resources:   []string{"networkpolicies"},
+							APIGroups:   []string{"*"},
+							Resources:   []string{"*"},
 							APIVersions: []string{"*"},
 						},
 						Operations: []admv1.OperationType{admv1.Create, admv1.Update, admv1.Delete},
 					},
-					ResourceNames: []string{"default"},
 				},
 			},
 		},
 		Validations: []admv1.Validation{
 			{
-				Expression: `"system:masters" in request.userInfo.groups || "system:serviceaccounts:kube-system" in request.userInfo.groups || "system:serviceaccounts:fleet-system" in request.userInfo.groups || "system:serviceaccounts:openshift-kube-controller-manager" in request.userInfo.groups`,
-				Message:    "Create, Update, or Delete operations on ARM managed resources is forbidden",
-				Reason:     &forbidden,
+				Expression: `(
+					request.userInfo.username == "aksService" ||
+					request.userInfo.username == "fleet-member-agent-sa")
```

Comment:
does adding this not break the Openshift case?

cc @nibooy

Reply:
What we found is that checking just the group is not enough as non Azure RBAC clusters will make the user kubeconfig in system:masters group, hence allowing users to delete and bypass this control.  By adding the clause that it must be `aksService`, which is the SA for relevant aks system components including fleet member agent (btw, I tested this), or `fleet-member-agent-sa` which is used by ARC, users should no longer be able to tamper with this control.